catalog/plugin/catalog-plugin-expirationdate/src/test/java/org/codice/ddf/catalog/plugin/expiration/TestExpirationDatePlugin.java

```diff
@@ -0,0 +1,597 @@
+        when(mockCreateRequest.getMetacards()).thenReturn(mockMetacards);
+
+        // Perform Test
+        expirationDatePlugin.process(mockCreateRequest);
+
+        // Verify
+        assertThat(mockCreateRequest.getMetacards().size(), is(numberOfMetacards));
+        assertNotNull(mockCreateRequest.getMetacards().get(0));
+        DateTime newExpirationDate = new DateTime(
+                mockCreateRequest.getMetacards().get(0).getExpirationDate());
+        // New exp date is 10 days after modified date
+        assertTrue(newExpirationDate.equals(MODIFIED_DATE.plusDays(days)));
+    }
+
+    /**
+     * Tests the process(CreateRequest createRequest) method.
+     * <p/>
+     * Don't overwrite the 'blank' expiration dates (multiple metacards) since the overwriteIfBlank option is not selected.
+     * <p/>
+     * offsetFromModifiedDate: 10
+     * overwriteIfBlank: false
+     * overwriteIfExists: false
+     *
+     * @throws PluginExecutionException
+     * @throws StopProcessingException
+     */
+    @Test
+    public void testProcessCreateRequestMetacardHasNoExpirationDateOverwriteIfBlankIsFalseOverwriteIfExistsIsFalseDaysOffsetFromModifiedDateIs10MultipleMetacards()
+            throws PluginExecutionException, StopProcessingException {
+
+        // Setup
+        ExpirationDatePlugin expirationDatePlugin = new ExpirationDatePlugin();
+
+        // Items configured via the admin console
+        int days = 10;
+        expirationDatePlugin.setOffsetFromModifedDate(days);
+        expirationDatePlugin.setOverwriteIfBlank(false);
+        expirationDatePlugin.setOverwriteIfExists(false);
+
+        CreateRequest mockCreateRequest = mock(CreateRequest.class);
+        int numberOfMetacards = 2;
+        List<Metacard> mockMetacards = createMockMetacardsWithNoExpirationDate(numberOfMetacards);
+        when(mockCreateRequest.getMetacards()).thenReturn(mockMetacards);
+
+        // Perform Test
+        expirationDatePlugin.process(mockCreateRequest);
+
+        // Verify
+        assertThat(mockCreateRequest.getMetacards().size(), is(numberOfMetacards));
+        for (Metacard metacard : mockCreateRequest.getMetacards()) {
+            assertNotNull(metacard);
+            // Don't overwrite 'blank' metacard exp dates
+            assertNull(metacard.getExpirationDate());
+        }
+    }
+
+    /**
+     * Tests the process(CreateRequest createRequest) method.
+     * <p/>
+     * Don't overwrite the 'blank' expiration dates (multiple metacards) since the overwriteIfBlank option is not selected.
+     * <p/>
+     * offsetFromModifiedDate: 10
+     * overwriteIfBlank: false
+     * overwriteIfExists: true
+     *
+     * @throws PluginExecutionException
+     * @throws StopProcessingException
+     */
+    @Test
+    public void testProcessCreateRequestMetacardHasNoExpirationDateOverwriteIfBlankIsFalseOverwriteIfExistsIsTrueDaysOffsetFromModifiedDateIs10MultipleMetacards()
```

Comment:
nit : we may want to shorten these method names, but that may just be my personal opinion